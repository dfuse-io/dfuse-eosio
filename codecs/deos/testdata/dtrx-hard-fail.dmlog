DMLOG START_BLOCK 40
DMLOG RAM_OP 0 7 deferred_trx remove deferred_trx_removed battlefield1 314758 -346
DMLOG CREATION_OP ROOT 0
DMLOG RAM_OP 0 battlefield1:battlefield1:member:...........aa table_row add primary_index_add battlefield1 314945 187
DMLOG DB_OP INS 0 battlefield1 battlefield1 battlefield1 member ...........aa 660000000000000066000000000000000000000000000000000000000000000022696e7365727465642062696c6c656420746f2063616c6c696e67206163636f756e74bf6e925d00000000
DMLOG DTRX_OP FAILED 0
DMLOG RLIMIT_OP ACCOUNT_USAGE UPD {"owner":"battlefield1","net_usage":{"last_ordinal":0,"value_ex":196157,"consumed":1},"cpu_usage":{"last_ordinal":0,"value_ex":450743,"consumed":1380},"ram_usage":314758}
DMLOG APPLIED_TRANSACTION 40 {"id":"76661ac8528b0155d2097846aba77da7461023741ae5ce01f7f7a8edb606e0f9","block_num":40,"block_time":"2019-09-30T21:08:15.500","producer_block_id":"0000002875422ebca92b7d64c93fa24dc53b0d9d0e806c41bdcb4a2563e90ea1","receipt":{"status":"hard_fail","cpu_usage_us":1379,"net_usage_words":0},"elapsed":0,"net_usage":0,"scheduled":true,"action_traces":[{"action_ordinal":1,"creator_action_ordinal":0,"closest_unnotified_ancestor_action_ordinal":0,"receipt":null,"receiver":"battlefield1","act":{"account":"battlefield1","name":"dbinstwo","authorization":[{"actor":"battlefield1","permission":"active"}],"data":{"account":"battlefield1","first":102,"second":100},"hex_data":"1052546ea998b33966000000000000006400000000000000"},"context_free":false,"elapsed":0,"console":"","trx_id":"76661ac8528b0155d2097846aba77da7461023741ae5ce01f7f7a8edb606e0f9","block_num":40,"block_time":"2019-09-30T21:08:15.500","producer_block_id":"0000002875422ebca92b7d64c93fa24dc53b0d9d0e806c41bdcb4a2563e90ea1","account_ram_deltas":[{"account":"battlefield1","delta":315}],"except":{"code":13,"name":"St9exception","message":"could not insert object, most likely a uniqueness constraint was violated","stack":[{"context":{"level":"warn","file":"wavm.cpp","line":0,"method":"call","hostname":"","thread_name":"thread-0","timestamp":"3333-12-31T00:01:02.345"},"format":"${what}: ","data":{"what":"could not insert object, most likely a uniqueness constraint was violated"}},{"context":{"level":"warn","file":"apply_context.cpp","line":0,"method":"exec_one","hostname":"","thread_name":"thread-0","timestamp":"3333-12-31T00:01:02.345"},"format":"pending console output: ${console}","data":{"console":""}}]},"error_code":null}],"account_ram_delta":{"account":"battlefield1","delta":-346},"except":{"code":13,"name":"St9exception","message":"could not insert object, most likely a uniqueness constraint was violated","stack":[{"context":{"level":"warn","file":"wavm.cpp","line":0,"method":"call","hostname":"","thread_name":"thread-0","timestamp":"3333-12-31T00:01:02.345"},"format":"${what}: ","data":{"what":"could not insert object, most likely a uniqueness constraint was violated"}},{"context":{"level":"warn","file":"apply_context.cpp","line":0,"method":"exec_one","hostname":"","thread_name":"thread-0","timestamp":"3333-12-31T00:01:02.345"},"format":"pending console output: ${console}","data":{"console":""}}]},"error_code":null}
DMLOG ACCEPTED_BLOCK 40 {"block_num":40,"dpos_proposed_irreversible_blocknum":40,"dpos_irreversible_blocknum":39,"active_schedule":{"version":0,"producers":[{"producer_name":"eosio","block_signing_key":"EOS5MHPYyhjBjnQZejzZHqHewPWhGTfQWSVTWYEhDmJu4SXkzgweP"}]},"blockroot_merkle":{"_active_nodes":["000000276e5289c65c429ce17acdea8603f7cac645e6a81c59ecd21a70253c7f","c03af6aa89c5c62dc22ac815d27e1d116dad676e16283298c09b5c53a928d031","3b81b5f386503956f0e983aa97c958168d5d144b007144a6be8fb8a6aa2bd3bf","83ae4f401c8afa997b555d5e6b0fc1b051febb1b9ed144b777706f231dc76716","c45c3b9c6de825eeb9339c5eb93e6a17e39ebb060edc96ac8fe60af87d54c26b"],"_node_count":39},"producer_to_last_produced":[["eosio",40]],"producer_to_last_implied_irb":[["eosio",39]],"block_signing_key":"EOS5MHPYyhjBjnQZejzZHqHewPWhGTfQWSVTWYEhDmJu4SXkzgweP","confirm_count":[],"id":"0000002875422ebca92b7d64c93fa24dc53b0d9d0e806c41bdcb4a2563e90ea1","header":{"timestamp":"2019-09-30T21:08:15.500","producer":"eosio","confirmed":0,"previous":"000000276e5289c65c429ce17acdea8603f7cac645e6a81c59ecd21a70253c7f","transaction_mroot":"0755749a80ce9bfff5d391149b146009737adec9943c0624d138500b4bc85958","action_mroot":"534ae58218df802e1258da817d7de0c9fd9dd90cba3be2d7a908e940face16d4","schedule_version":0,"header_extensions":[],"producer_signature":"SIG_K1_KatMYbC5T1H43VX7o3MqXSCrwzGGiHoA26XUtWtazSCc18c1bQ9CFF9YrJTTLRKxFLWs6S5g2NZE9HikVjhr1YjCWdS1KB"},"pending_schedule":{"schedule_lib_num":0,"schedule_hash":"2f74e78d3382fdadcbbdb7b111ca91efeea4109d59ac94945951fca6c498aef6","schedule":{"version":0,"producers":[]}},"activated_protocol_features":{"protocol_features":[]},"block":{"timestamp":"2019-09-30T21:08:15.500","producer":"eosio","confirmed":0,"previous":"000000276e5289c65c429ce17acdea8603f7cac645e6a81c59ecd21a70253c7f","transaction_mroot":"0755749a80ce9bfff5d391149b146009737adec9943c0624d138500b4bc85958","action_mroot":"534ae58218df802e1258da817d7de0c9fd9dd90cba3be2d7a908e940face16d4","schedule_version":0,"header_extensions":[],"producer_signature":"SIG_K1_KatMYbC5T1H43VX7o3MqXSCrwzGGiHoA26XUtWtazSCc18c1bQ9CFF9YrJTTLRKxFLWs6S5g2NZE9HikVjhr1YjCWdS1KB","transactions":[{"status":"hard_fail","cpu_usage_us":1379,"net_usage_words":0,"trx":[0,"76661ac8528b0155d2097846aba77da7461023741ae5ce01f7f7a8edb606e0f9"]}],"block_extensions":[]},"validated":false}