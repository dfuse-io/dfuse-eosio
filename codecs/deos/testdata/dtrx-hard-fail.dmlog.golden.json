{
  "id": "0000002875422ebca92b7d64c93fa24dc53b0d9d0e806c41bdcb4a2563e90ea1",
  "number": 40,
  "header": {
    "timestamp": "2019-09-30T21:08:15.500Z",
    "producer": "eosio",
    "previous": "000000276e5289c65c429ce17acdea8603f7cac645e6a81c59ecd21a70253c7f",
    "transactionMroot": "0755749a80ce9bfff5d391149b146009737adec9943c0624d138500b4bc85958",
    "actionMroot": "534ae58218df802e1258da817d7de0c9fd9dd90cba3be2d7a908e940face16d4"
  },
  "producerSignature": "SIG_K1_KatMYbC5T1H43VX7o3MqXSCrwzGGiHoA26XUtWtazSCc18c1bQ9CFF9YrJTTLRKxFLWs6S5g2NZE9HikVjhr1YjCWdS1KB",
  "transactions": [
    {
      "id": "76661ac8528b0155d2097846aba77da7461023741ae5ce01f7f7a8edb606e0f9",
      "status": "TRANSACTIONSTATUS_HARDFAIL",
      "cpuUsageMicroSeconds": 1379
    }
  ],
  "transactionCount": 1,
  "dposProposedIrreversibleBlocknum": 40,
  "dposIrreversibleBlocknum": 39,
  "blockrootMerkle": {
    "nodeCount": 39,
    "activeNodes": [
      "000000276e5289c65c429ce17acdea8603f7cac645e6a81c59ecd21a70253c7f",
      "c03af6aa89c5c62dc22ac815d27e1d116dad676e16283298c09b5c53a928d031",
      "3b81b5f386503956f0e983aa97c958168d5d144b007144a6be8fb8a6aa2bd3bf",
      "83ae4f401c8afa997b555d5e6b0fc1b051febb1b9ed144b777706f231dc76716",
      "c45c3b9c6de825eeb9339c5eb93e6a17e39ebb060edc96ac8fe60af87d54c26b"
    ]
  },
  "producerToLastProduced": [
    {
      "name": "eosio",
      "lastBlockNumProduced": 40
    }
  ],
  "producerToLastImpliedIrb": [
    {
      "name": "eosio",
      "lastBlockNumProduced": 39
    }
  ],
  "confirmCount": [
  ],
  "pendingSchedule": {
    "scheduleHash": "2f74e78d3382fdadcbbdb7b111ca91efeea4109d59ac94945951fca6c498aef6",
    "scheduleV1": {

    },
    "scheduleV2": {

    }
  },
  "activatedProtocolFeatures": {
    "protocolFeatures": [
    ]
  },
  "transactionTraces": [
    {
      "id": "76661ac8528b0155d2097846aba77da7461023741ae5ce01f7f7a8edb606e0f9",
      "blockNum": "40",
      "blockTime": "2019-09-30T21:08:15.500Z",
      "producerBlockId": "0000002875422ebca92b7d64c93fa24dc53b0d9d0e806c41bdcb4a2563e90ea1",
      "receipt": {
        "status": "TRANSACTIONSTATUS_HARDFAIL",
        "cpuUsageMicroSeconds": 1379
      },
      "scheduled": true,
      "actionTraces": [
        {
          "receiver": "battlefield1",
          "action": {
            "account": "battlefield1",
            "name": "dbinstwo",
            "authorization": [
              {
                "actor": "battlefield1",
                "permission": "active"
              }
            ],
            "jsonData": "{\"account\":\"battlefield1\",\"first\":102,\"second\":100}",
            "rawData": "1052546ea998b33966000000000000006400000000000000"
          },
          "transactionId": "76661ac8528b0155d2097846aba77da7461023741ae5ce01f7f7a8edb606e0f9",
          "blockNum": "40",
          "producerBlockId": "0000002875422ebca92b7d64c93fa24dc53b0d9d0e806c41bdcb4a2563e90ea1",
          "blockTime": "2019-09-30T21:08:15.500Z",
          "accountRamDeltas": [
            {
              "account": "battlefield1",
              "delta": "315"
            }
          ],
          "exception": {
            "code": 13,
            "name": "St9exception",
            "message": "could not insert object, most likely a uniqueness constraint was violated",
            "stack": [
              {
                "context": {
                  "level": "warn",
                  "file": "wavm.cpp",
                  "method": "call",
                  "threadName": "thread-0",
                  "timestamp": "3333-12-31T00:01:02.345Z"
                },
                "format": "${what}: ",
                "data": "7b2277686174223a22636f756c64206e6f7420696e73657274206f626a6563742c206d6f7374206c696b656c79206120756e697175656e65737320636f6e73747261696e74207761732076696f6c61746564227d"
              },
              {
                "context": {
                  "level": "warn",
                  "file": "apply_context.cpp",
                  "method": "exec_one",
                  "threadName": "thread-0",
                  "timestamp": "3333-12-31T00:01:02.345Z"
                },
                "format": "pending console output: ${console}",
                "data": "7b22636f6e736f6c65223a22227d"
              }
            ]
          },
          "actionOrdinal": 1
        }
      ],
      "exception": {
        "code": 13,
        "name": "St9exception",
        "message": "could not insert object, most likely a uniqueness constraint was violated",
        "stack": [
          {
            "context": {
              "level": "warn",
              "file": "wavm.cpp",
              "method": "call",
              "threadName": "thread-0",
              "timestamp": "3333-12-31T00:01:02.345Z"
            },
            "format": "${what}: ",
            "data": "7b2277686174223a22636f756c64206e6f7420696e73657274206f626a6563742c206d6f7374206c696b656c79206120756e697175656e65737320636f6e73747261696e74207761732076696f6c61746564227d"
          },
          {
            "context": {
              "level": "warn",
              "file": "apply_context.cpp",
              "method": "exec_one",
              "threadName": "thread-0",
              "timestamp": "3333-12-31T00:01:02.345Z"
            },
            "format": "pending console output: ${console}",
            "data": "7b22636f6e736f6c65223a22227d"
          }
        ]
      },
      "ramOps": [
        {
          "operation": "OPERATION_DEFERRED_TRX_REMOVED",
          "payer": "battlefield1",
          "delta": "-346",
          "usage": "314758",
          "namespace": "NAMESPACE_DEFERRED_TRX",
          "uniqueKey": "7",
          "action": "ACTION_REMOVE"
        }
      ],
      "rlimitOps": [
        {
          "operation": "OPERATION_UPDATE",
          "accountUsage": {
            "owner": "battlefield1",
            "netUsage": {
              "valueEx": "196157",
              "consumed": "1"
            },
            "cpuUsage": {
              "valueEx": "450743",
              "consumed": "1380"
            },
            "ramUsage": "314758"
          }
        }
      ]
    }
  ],
  "transactionTraceCount": 1,
  "executeInputActionCount": 1,
  "executedTotalActionCount": 1,
  "blockSigningKey": "EOS5MHPYyhjBjnQZejzZHqHewPWhGTfQWSVTWYEhDmJu4SXkzgweP",
  "activeScheduleV1": {
    "producers": [
      {
        "accountName": "eosio",
        "blockSigningKey": "EOS5MHPYyhjBjnQZejzZHqHewPWhGTfQWSVTWYEhDmJu4SXkzgweP"
      }
    ]
  }
}